data/transactions/logic/eval.go

```diff
@@ -1678,6 +1845,124 @@ func opSubstring3(cx *evalContext) {
 	cx.stack = cx.stack[:prev]
 }
 
+func opGetBit(cx *evalContext) {
+	last := len(cx.stack) - 1
+	prev := last - 1
+	idx := cx.stack[last].Uint
+	target := cx.stack[prev]
+
+	var bit uint64
+	if target.argType() == StackUint64 {
+		if idx > 63 {
+			cx.err = errors.New("getbit index > 63 with with Uint")
+			return
+		}
+		mask := uint64(1) << idx
+		bit = (target.Uint & mask) >> idx
+	} else {
+		// indexing into a byteslice
+		byteIdx := idx / 8
+		if byteIdx >= uint64(len(target.Bytes)) {
+			cx.err = errors.New("getbit index beyond byteslice")
+			return
+		}
+		byteVal := target.Bytes[byteIdx]
+
+		bitIdx := idx % 8
+		// We saying that bit 9 (the 10th bit), for example,
+		// is the 2nd bit in the second byte, and that "2nd
+		// bit" here means almost-highest-order bit, because
+		// we're thinking of the bits in the byte itself as
+		// being big endian. So this looks "reversed"
```

Comment:
Okay yeah it's weird that uint64 is little-bit-endian but byte-array is the other way around. What if we treated uint64 as big endian [8]byte ? Or either way just consistent.

Reply:
The trouble is that I just don't think you can get around the fact that people expect "bit 0" to mean "lower order bit" of an integer, and for "byte 0" (and therefore "bit 0") to mean the "left-most" byte (or bit) in a string.

I convinced myself it was reasonable. The direction depends on the type being indexed. I'm amenable to counter-arguments though.  Have a look at the unit test to be convinced though.  I think when you see the operations in action, they do what you expect.

Reply:
Here's what I find convincing
```
	// bit 10 is the 3rd bit (from the high end) in the second byte
	testAccepts(t, "byte 0xfff0; int 10; getbit; int 1; ==", 3)
	testAccepts(t, "byte 0xfff0; int 12; getbit; int 0; ==", 3)
```

You really want the 10 to see the "f" part of the third byte, and 12 to see the 0 part.

Reply:
ok but
```
int 0xfff0; int 12; getbit; int 0; ==
```
should do the same thing?

Reply:
I would say no.  I think that everyone has decided that integer bit indexing is based on math, rather than layout.  In C, whether you're on a big-endian or little-endian machine, 0xfff0 & (1 << 2),  is going to take a 0 bit from that low-order nybble.  I think that's what people think of as bit indexing.  That expression is "give me bit 2", and it yields the same thing on x86 and sparc.